### PR TITLE
Promote from testing-dashboard-service-uptime to main

### PR DESCRIPTION
**Pull request by social-surveys-blaise-concourse**
Auto generated by concourse:

Promote from testing-dashboard-service-uptime to main

### CODE REVIEW

File: src/app.tsx [hunk 1]
```diff
@@ -125,6 +181,10 @@ export default class App extends Component<unknown, AppState> {
                         {this.loadingPanel()}
                         {this.reportTable()}
                         {this.completedCaseDefinition()}
+                        <h2 className="ons-u-mt-m">Service Health Check Information</h2>
```
Comment (Enigma-cloud) [inline src/app.tsx:184]: Very trivial and not really important but it might be good to use lowercase for the rest (to match the "Completed case information" above it)

File: src/app.test.tsx [hunk 2]
```diff
@@ -43,7 +47,8 @@ describe("App", () => {
```
Comment (tomphp) [inline src/app.test.tsx:47]: This tests needs a `getMonitoringMock.mockImplementation(() => Promise.resolve([]));` to fix the snapshot.

File: src/app.test.tsx [hunk 3]
```diff
@@ -86,4 +93,21 @@ describe("App", () => {
...
+    it("renders an info panel", async () => {
+      getCaseCompletionReportMock.mockImplementation(() => Promise.resolve(caseCompletionReport));
+      getQuestionnairesMock.mockImplementation(() => Promise.resolve([]));
+      getMonitoringMock.mockImplementation(() => Promise.resolve([]));
+
+      render(
+        <App />
+      );
+
+      await waitFor(() => {
```
Comment (tomphp) [inline src/app.test.tsx:107]: This block  *could* be replaced by (i.e. no need for `waitFor`)
```typescript
await expect(screen.findByText("No Uptime checks data.")).toBeDefined();
```

File: src/app.tsx [hunk 4]
```diff
@@ -31,22 +37,44 @@ export default class App extends Component<unknown, AppState> {
...
+            console.log(uptimeChecks);
+            this.setState({
+                uptimeChecks: uptimeChecks,
+                uptimeChecksLoading : false
+            });
+        }).catch((reason: unknown) => {
+            console.error(reason);
+            this.setState({
+                uptimeChecks: [],
+                uptimeChecksLoading : true
```
Comment (tomphp) [inline src/app.tsx:73]: This should `uptimeChecksErrored` - indicates a missing test.

File: src/app.tsx [hunk 5]
```diff
@@ -79,6 +118,13 @@ export default class App extends Component<unknown, AppState> {
         return undefined;
     }
 
+    UptimeChecksloadingPanel(): ReactElement | undefined {
```
Comment (tomphp) [inline src/app.tsx:121]: All other functions start with a lowercase letter 😁

File: src/app.tsx [hunk 6]
```diff
@@ -31,22 +37,44 @@ export default class App extends Component<unknown, AppState> {
         this.state = {
             questionnaires: [],
             loading: true,
-            errored: false
+            errored: false,
```
Comment (tomphp) [inline src/app.tsx:40]: Should these now be `questionnaireErrored` and `questionnaireLoading` (same for the method names below too)

File: src/app.tsx [hunk 7]
```diff
@@ -65,12 +93,23 @@ export default class App extends Component<unknown, AppState> {
...
+        return await getMonitoring();
+    }
+
     errorPanel(): ReactElement | undefined {
         if (this.state.errored) {
             return <ONSPanel status="error">Failed to get questionnaires.</ONSPanel>;
         }
         return undefined;
     }
+    uptimeChecksErrorPanel(): ReactElement | undefined {
```
Comment (tomphp) [inline src/app.tsx:106]: Missing newline above this method

File: src/components/monitoringUptimeChecks.test.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        render(
+            <table>
+                <tbody>
+                    <MonitoringUptimeChecks hostname={"dev-bts.social-surveys.gcp.onsdigital.uk"} eurBelgium={"success"} apacSingapore={"error"} northAmerica={"success"} southAmerica={"requestFailed"}/>
+                </tbody>
+            </table>
+        );
+
+        await waitFor(() => {
+            expect(screen.getByTestId("uptimecheck-dev-bts.social-surveys.gcp.onsdigital.uk").textContent).toEqual("dev-bts.social-surveys.gcp.onsdigital.uk");
```
Comment (tomphp) [inline src/components/monitoringUptimeChecks.test.tsx:21]: This could use `findByTestId` instead

File: src/components/monitoringUptimeChecksTable.test.tsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+import { getMonitoring } from "../client/monitoring";
+import { MonitoringDataModel } from "../server/monitoringDataModel";
+
+import MonitoringUptimeChecksTable from "./monitoringUptimeChecksTable";
+
+const getMonitoringMock = getMonitoring as jest.Mock<Promise<MonitoringDataModel[]>>;
+
+
+describe("Monitoring Uptime Checks Table", () => {
+
```
Comment (tomphp) [inline src/components/monitoringUptimeChecksTable.test.tsx:19]: Unnecessary whitespace

File: src/components/monitoringUptimeChecksTable.test.tsx [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+        getMonitoringMock.mockImplementation(() => Promise.resolve(mockHealthCheckList));
+
+        render(
+            <MonitoringUptimeChecksTable monitoringData={mockHealthCheckList} />
+        );
+
+        expect(screen.queryByText(/Loading/i)).toBeDefined();
+
+        await waitFor(() => {
+            expect(screen.getByTestId("uptimecheck-dev-bts.social-surveys.gcp.onsdigital.uk").textContent).toEqual("dev-bts.social-surveys.gcp.onsdigital.uk");
```
Comment (tomphp) [inline src/components/monitoringUptimeChecksTable.test.tsx:35]: Could use `findBy*` instead

File: src/server/googleCloudMonitoringApi/googleMonitoringApi.ts [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+        try{
+        const request = {
+            parent: this.uptimeClient.projectPath(this.projectId)
+        };
+
+        // Retrieves an uptime check config with hostnames-services like dqs, bus, bum, dashboard etc
+        const [uptimeCheckConfigs] = await this.uptimeClient.listUptimeCheckConfigs(request);
+
+        return uptimeCheckConfigs;
+    }catch(error: any)
```
Comment (tomphp) [inline src/server/googleCloudMonitoringApi/googleMonitoringApi.ts:22]: Indentation and spacing is a bit of a mess

File: src/server/googleCloudMonitoringApi/monitoring.test.ts [hunk 12]
```diff
@@ -0,0 +1,111 @@
+import {getMonitoringUptimeCheckTimeSeries} from "./monitoring";
+
+const mockGoogleMonitoring = {
+    getUptimeChecksConfigs: jest.fn(),
+    listTimeSeries: jest.fn(),
+};
+
+describe("Get all uptime checks from API", () => {
+    it("should return success statuses", async () => {
+        mockGoogleMonitoring.getUptimeChecksConfigs.mockReturnValue(
```
Comment (tomphp) [inline src/server/googleCloudMonitoringApi/monitoring.test.ts:10]: I think this could be:

```typescript
mockGoogleMonitoring.getUptimeChecksConfigs.mockResolvedValue(
  [ {monitoredResource: {labels: {host: "example-host"}}} ]
);
```

File: src/server/googleCloudMonitoringApi/monitoring.test.ts [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+                    {"region": "eur-belgium", "status": "success"},
+                    {"region": "apac-singapore", "status": "success"},
+                    {"region": "usa-oregon", "status": "success"},
+                    {"region": "sa-brazil-sao_paulo", "status": "success"}
+                ]
+            }
+        ]);
+    });
+
+     // TODO: write more tests
```
Comment (tomphp) [inline src/server/googleCloudMonitoringApi/monitoring.test.ts:36]: Don't need this any more 😁

File: src/server/googleCloudMonitoringApi/monitoring.ts [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+export type GetUptimeChecksConfigResult = protos.google.monitoring.v3.IUptimeCheckConfig[];
+export type ListTimeSeriesResult = protos.google.monitoring.v3.ITimeSeries[];
+
+export interface GoogleMonitoring {
+    getUptimeChecksConfigs(): Promise<GetUptimeChecksConfigResult>;
+    listTimeSeries(filter: string, startTime: number, endTime: number): Promise<ListTimeSeriesResult>;
+}
+
+export async function getMonitoringUptimeCheckTimeSeries(
+
```
Comment (tomphp) [inline src/server/googleCloudMonitoringApi/monitoring.ts:16]: Formatting

File: src/app.tsx (hunk 1 above)
Comment (SidraJaved) [reply src/app.tsx:184]: Done

File: src/app.test.tsx (hunk 2 above)
Comment (SidraJaved) [reply src/app.test.tsx:47]: It helped fix the snapshot too!

File: src/app.test.tsx (hunk 3 above)
Comment (SidraJaved) [reply src/app.test.tsx:107]: Needs discussion!

File: src/app.tsx (hunk 4 above)
Comment (SidraJaved) [reply src/app.tsx:73]: Changes made to update uptimeChecksErrored and missing test has been added in app.test.tsx

File: src/app.tsx (hunk 5 above)
Comment (SidraJaved) [reply src/app.tsx:121]: Done!

File: src/app.tsx (hunk 6 above)
Comment (SidraJaved) [reply src/app.tsx:40]: Done, state attributes as well as method names have been updated to show that they relate to questionnaires.

File: src/app.tsx (hunk 7 above)
Comment (SidraJaved) [reply src/app.tsx:106]: ✅

File: src/app.tsx (hunk 1 above)
Comment (SidraJaved) [reply src/app.tsx:184]: ✅

File: src/components/monitoringUptimeChecks.test.tsx (hunk 8 above)
Comment (SidraJaved) [reply src/components/monitoringUptimeChecks.test.tsx:21]: Needs discussion

File: src/components/monitoringUptimeChecksTable.test.tsx (hunk 9 above)
Comment (SidraJaved) [reply src/components/monitoringUptimeChecksTable.test.tsx:19]: Done and added missing test, also done some formatting (LINTING 😁 )...

File: src/components/monitoringUptimeChecksTable.test.tsx (hunk 10 above)
Comment (SidraJaved) [reply src/components/monitoringUptimeChecksTable.test.tsx:35]: Discussion please!

File: src/app.test.tsx (hunk 3 above)
Comment (Enigma-cloud) [reply src/app.test.tsx:107]: I think you might need to put `await` inside the expect statement like:
`expect(await screen.findByText("No Uptime checks data.")).toBeDefined();`

File: src/server/googleCloudMonitoringApi/googleMonitoringApi.ts (hunk 11 above)
Comment (SidraJaved) [reply src/server/googleCloudMonitoringApi/googleMonitoringApi.ts:22]: ✅

File: src/server/googleCloudMonitoringApi/monitoring.test.ts (hunk 12 above)
Comment (SidraJaved) [reply src/server/googleCloudMonitoringApi/monitoring.test.ts:10]: Done

File: src/components/monitoringUptimeChecks.test.tsx (hunk 8 above)
Comment (SidraJaved) [reply src/components/monitoringUptimeChecks.test.tsx]: All done and sorted after discussion :)
